code/modules/reagents/chemistry/reagents/food.dm

```diff
@@ -275,6 +275,44 @@
 		M.adjustBrainLoss(1)
 	..()
 
+/datum/reagent/consumable/sodiumchloride/reaction_mob(mob/living/M, method=TOUCH, volume)
+	if(method == TOUCH || method == INGEST)
```

Comment:
I'd probably remove this line as the method will always be either of these, making it redundant. You're also checking these things later on in the proc.